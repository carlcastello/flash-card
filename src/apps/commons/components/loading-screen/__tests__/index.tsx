import React from "react";
import { shallow } from "enzyme";

import { LoadingScreen } from "../index";
import toJson from "enzyme-to-json";
import { LinearProgress } from "@material-ui/core";

describe("", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <LoadingScreen
        classes={{
          boxContainer: "box-container",
          boxContent: "box-content",
          typography: "typography",
          linearProgressBar: "linear-progress-bar"
        }}/>
    )
  });


  it("displays the loading screen", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(LinearProgress)).toHaveLength(1);
  });
});