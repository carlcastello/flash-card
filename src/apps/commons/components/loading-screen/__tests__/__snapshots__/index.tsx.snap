// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the loading screen 1`] = `
<Styled(MuiBox)
  className="box-container"
>
  <Styled(MuiBox)
    className="box-content"
  >
    <WithStyles(ForwardRef(Typography))
      className="typography"
      variant="h4"
    />
    <WithStyles(ForwardRef(LinearProgress))
      className="linear-progress-bar"
      color="secondary"
    />
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
